### ISSUE TITLE
Default dict value in JSONField causes non-valid JSON string stored in db.

### ISSUE THREAD

**Issue by nebril** [skorokithakis/django-annoying#32]
I have a model with JSONField with default value set to constant dict value.

The default value was stored in db as non-valid json.

I was able to work around this by changing this line (in my model):
`visual_config = JSONField(default=DEFAULT_PARAMETERS['visual_config'], blank=False, null=False)`
With this:
`visual_config = JSONField(default=lambda: CustomerConfig.DEFAULT_PARAMETERS['visual_config'], blank=False, null=False)`

Notice that I added default value as lambda function.

This is caused by django default `Field.get_default` method, which runs `force_text` method on the default value if it is not callable:
```    
def get_default(self):
        """
        Returns the default value for this field.
        """
        if self.has_default():
            if callable(self.default):
                return self.default()
            return force_text(self.default, strings_only=True)
        if (not self.empty_strings_allowed or (self.null and
                   not connection.features.interprets_empty_strings_as_nulls)):
            return None
        return ""
```

My idea for a fix would be to override the get_default method in JSONField.

Cheers,
Maciej

**Comment by ketanbhatt** [comment]
A valid JSON would be something like {"key": "value"}
But if we store a static dict as the default, it gets stored as {'key': 'value'} which becomes an Invalid JSON.

Any workaround for this?

**Comment by skorokithakis** [comment]
Hmm, this definitely shouldn't be the case. Can you post a small sample that reproduces the problem?

**Comment by ketanbhatt** [comment]
I found a workaround. Doing a `json.dumps` gets rid of the issue.

My dict was like:

```python
mydic = { "key": "value"}
```

I stored it as valid json by:

 ```python
class NotificationTemplate(models.Model):
    extra = JSONField(default=json.dumps(NOTIFICATION_TEMPLATE_EXTRA)))
```

Remove the `json.dumps` it gets stored as:

```python
{ 'key': 'value'}
```

which is invalid

**Comment by skorokithakis** [comment]
That's very odd. I'll try to have a look and see why this happens, thanks.

**Comment by skorokithakis** [comment]
Fixed in 6912356, as @nebril suggested. Thank you all for the report!
